107-tee-boot-attestation.md

```diff
@@ -0,0 +1,509 @@
+│Processor│       │           +            ├───►├────────────┼──►│Registrar│
+└────┬────┘       │      CVM Integrity     │    │ CVM        │   └─────────┘
+     │            │      public Key        │    │ Integrity  │              
+     │            │           +            │    │ public Key │              
+     │            │      CHIP_ID           │    ├────────────┤              
+     │            │      (or TDX/CCA/etc   │    │ Encrypted  │              
+     │            │       equivalent)      │    │ vTPM State │              
+     │            │                        │    │ Key        │              
+     │            │      Into REPORT_DATA  │    └────────────┘              
+     │            └────────────────────────┘                                
+     │                          ▲                                           
+     └──────────────────────────┘                                           
+        4. Attestation Report                                               
+```
+
+- Registrar will attest the report (with TEE certificates) and report a
+  pass/fail.
+
+UPON SUCCESSFUL ATTESTATION
+- Registrar will decrypt the vTPM State Key that was previously encrypted
+  with its TEE Attestation Key.
+- As mentioned before, SVSM provides a CVM Integrity Key to prevent the
+  untrusted host from spying on decrypted secrets. The vTPM State Key is
+  encrypted with this key and sent back to SVSM to decrypt.
+
+```
+                    3. Decrypt      4. Encrypt        
+                       vTPM State      vTPM State Key 
+                  ┌──► Key w/ TEE  ──► w/             
+                  │    Attestation     CVM Integrity  
+                  │    Key             Key            
+             ┌────┴────┐                   │          
+             │Registrar│                   │          
+             └─┬─────▲─┘                   │          
+               │     │                     │          
+1. Attestation │     │2. Pass/Fail         │          
+   Report      │     │                     │    ┌────┐
+        +      │     │                     └───►│SVSM│
+   Certificates│     │                          └────┘
+               │     │                                
+            ┌──▼─────┴──┐                             
+            │TEE        │                             
+            │Attestation│                             
+            │Verifier   │                             
+            └───────────┘                             
+```
+
+- As this is the "first phase" of registration, the Registrar will store the
+  UUID in its database:
+
+```
+┌─────────┐ Agent UUID  ┌───────────┐
+│Registrar├────────────►│ Registrar │
+└─────────┘             │    DB     │
+                        └───────────┘
+```
+
+- SVSM will decrypt its vTPM State Key with its private CVM Integrity Key.
+- SVSM will decrypt vTPM state with vTPM State Key.
+- SVSM will derive key from vTPM state and use to unlock rootfs.
```

Comment:
I'd remove this line. The rootfs will be unlocked from the guest OS. Kernel and initird are not encrypted (but measured), so sytemd is already able to unlock the rootfs using a key sealed with the TPM.